lib/charms/data_platform_libs/v0/upgrade.py

```diff
@@ -532,10 +532,11 @@ def __init__(
         self.framework.observe(
             getattr(self.charm.on, "pre_upgrade_check_action"), self._on_pre_upgrade_check_action
         )
-        if self.substrate == "k8s":
-            self.framework.observe(
-                getattr(self.charm.on, "resume_upgrade_action"), self._on_resume_upgrade_action
-            )
+        # FIXME: this is disabled as we don't need the functionality. Should be added as an arg to the lib
```

Comment:
Why we don't need this feature? In general I would like to keep the UX the same across products (whenever this is not prevented by the product, but this does not seem to be the case here AFAIK)

Reply:
Yeah, so this is definitely polemic and I'm fine with removing this change if we disagree.

As it stands right now, VM + K8s upgrades (all DP products) have very different UX.
This change prioritises consistency 'within' a product, i.e making Kafka + ZooKeeper substrate upgrades identical.
The feature itself is a good idea, but until we have a consistent flow across substrates, I'm not sure it's worth including, as it's a fair 'speed-bump' to an upgrade generally.

I am unsure how often we are likely to want this behavior, given that we've already done upgrades in production without it, and it never came up. The reason that the feature exists in the first place, is on the theory that one can spin up a 'sacrificial' unit to test the upgrade on. I don't think it's that simple to do in production environments under load, and CERTAINLY not on VM (where discussions on machine resourcing can be time-consuming).

If I were to propose a flow that made sense for both, I would suggest:
1. Flag set during `pre-upgrade-check` on how we want the upgrade to progress. `automated` or `pausing`
2. If `pausing`, pause upgrade after EVERY unit, not just the first
3. (somewhat unrelated) Use `upgrade-granted` event on both substrates

Reply:
I honestly like the flow you are proposing, but it is not just **our** call. I would really prioritize consistent UX across products, here. 

Unfortunately, K8s and VM substrates are different and it is fair to acknowledge that. Whenever possible, I would like to keep the UX as consistent as possible between them, but I also understand that sometimes this will not be feasible. But - on the other hand - as Data Platform, we should have a consistent way to approach both, although this can lead to small differences between the two substrates (but because they have different mechanics)

Again, I'm not again this change or changing the UX. I agree with you on the flow. But I would just like to keep this one on hold until we have a decision on this shared with the other teams.